### PR TITLE
[W 9][CS2103AUG2017-F11-B2]AN JIASHENG

### PR DESCRIPTION
**Pull request by AngularJiaSheng**
Web Links for users

### CODE REVIEW

**Comment by hansiang93** [comment]
OH MY i thought you'd just parse them as a list of strings T.T

Its ok, weblinks work as well. I'd have to change some stuff on the "BrowserPanel" side of things.

File: src/main/java/seedu/address/model/WebLink/WebLink.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a WebLink in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidWebLink(String)}
+ */
+public class WebLink {
+
+    public static final String MESSAGE_WEB_LINK_CONSTRAINTS = "it should be a web link you want to stalk.";
+    public static final String WEB_LINK_VALIDATION_REGEX = "^\\S+";
```
Comment (hansiang93) [inline src/main/java/seedu/address/model/WebLink/WebLink.java:14]: Might want to make this more defined....... for future update.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 2]
```diff
@@ -24,22 +25,22 @@
             return new Person[] {
                 new Person(new Name("Alex Yeoh"), new Phone("87438807"), new Email("alexyeoh@example.com"),
                     new Address("Blk 30 Geylang Street 29, #06-40"), new Remark(""),
-                    getTagSet("friends")),
+                    getTagSet("friends"), getWebLinkSet("https://www.facebook.com/Phua.Han.Siang")),
```
Comment (hansiang93) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:28]: ......

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -26,6 +28,7 @@
 
     private final UniquePersonList persons;
     private final UniqueTagList tags;
+    private final UniqueWebLinkList webLinks;
```
Comment (hansiang93) [inline src/main/java/seedu/address/model/AddressBook.java:31]: Should the webLinks exist as a common list in the Addressbook, or as a list within the "Person" class?

File: src/main/java/seedu/address/model/AddressBook.java [hunk 4]
```diff
@@ -149,6 +160,37 @@ private void syncMasterTagListWith(UniquePersonList persons) {
...
+    private void syncMasterWebLinkListWith(Person person) {
+        final UniqueWebLinkList personWebLinks = new UniqueWebLinkList(person.getWebLinks());
+        webLinks.mergeFrom(personWebLinks);
+
+        // Create map with values = webLink object references in the master list
+        // used for checking person webLink references
+        final Map<WebLink, WebLink> masterWebLinkObjects = new HashMap<>();
+        webLinks.forEach(webLink -> masterWebLinkObjects.put(webLink, webLink));
+
+        // Rebuild the list of person tags to point to the relevant webLink in the master webLink list.
```
Comment (hansiang93) [inline src/main/java/seedu/address/model/AddressBook.java:176]: is there a need for a "master webLink list"?

File: src/main/java/seedu/address/model/WebLink/WebLink.java (hunk 1 above)
Comment (AngularJiaSheng) [reply src/main/java/seedu/address/model/WebLink/WebLink.java:14]: hahaha okayokay

Comment (AngularJiaSheng) [reply src/main/java/seedu/address/model/WebLink/WebLink.java:14]: okay the regex i set as such cos, really have too many permutations, https://, http, www, or no www. I didnt see a point in setting a restrictive regex.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java (hunk 2 above)
Comment (AngularJiaSheng) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:28]: XD

Comment (AngularJiaSheng) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:28]: It came back to haunt me when as I was doing testing, as it keep showing up for no reasons as actual outcome hahahaha...

File: src/main/java/seedu/address/model/AddressBook.java (hunk 4 above)
Comment (AngularJiaSheng) [reply src/main/java/seedu/address/model/AddressBook.java:176]: yah actually u r right, hmmmmmm, i added it the other day in cos for some reason my links were no showing up on the UI and i thought this was the problem

**Review by bladerail — APPROVED** [review]
Checked and approved. Merging

**Pull request merged**